### PR TITLE
Slab hash final implementation

### PR DESCRIPTION
**Pull request by bagrorg**

### CODE REVIEW

**Comment by bagrorg** [comment]
@kurapov-peter

File: hash/slab_hash_build.cpp [hunk 1]
```diff
@@ -45,8 +48,9 @@ void SlabHashBuild::_run(const size_t buf_size, Meter &meter) {
                     [intel::reqd_sub_group_size(SlabHash::SUBGROUP_SIZE)]] {
                size_t ind = it.get_group().get_id();
 
-               SlabHash::SlabHashTable<uint32_t, uint32_t,
-                                       SlabHash::DefaultHasher<5, 11, 1031>>
+               SlabHash::SlabHashTable<
+                   uint32_t, uint32_t,
+                   SlabHash::DefaultHasher<242792921, 653019598, 2147483647>>
```
Comment (kurapov-peter) [inline hash/slab_hash_build.cpp:53]: magic numbers

**Review by kurapov-peter — APPROVED** [review]
Please replace magic constants if possible, otherwise LGTM

**Comment by kurapov-peter** [comment]
@bagrorg, let's merge this one, please resolve conflicts

File: common/common.cpp [hunk 2]
```diff
@@ -4,6 +4,21 @@
 #include <set>
 
 namespace helpers {
+std::vector<uint32_t> make_unique_random(size_t size) {
+  constexpr size_t size_multiplier = 10;
+
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<int> dist(
+      1, std::min((long)(size * size_multiplier), (long)((uint32_t)-1)));
```
Comment (kurapov-peter) [inline common/common.cpp:13]: std::numeric_limits<uint32_t>::max() - 1

Comment (kurapov-peter) [reply common/common.cpp:13]: better remove this change

**Pull request merged**